doc/contribute/development_workflow.rst

```diff
@@ -0,0 +1,630 @@
+installation of Git. If you have Git set up already, skip to the :ref:`workflow` section.
+
+Installation and global setup
+#############################
+
+1. `Install Git <http://git-scm.com/book/en/v2/Getting-Started-Installing-Git>`_.
+2. Introduce yourself to Git:
+
+.. code::
+
+   git config --global user.email you@yourdomain.example.com
+   git config --global user.name "Your Name Comes Here"
+
+Setting up your GitHub account
+##############################
+
+The NEST source code is hosted in a public repository on
+`GitHub <https://github.com/nest/nest-simulator>`_. If you don’t have a GitHub
+account already, please create one.
+
+You then need to configure your account to allow write access - please see the
+`article on generating SSH keys <http://help.github.com/articles/generating-ssh-keys>`_
+on the `GitHub help website <https://help.github.com/>`_.
+
+.. _fork:
+
+Making your own copy (fork) of NEST
+###################################
+
+This needs to be done only once. The instructions here are very similar to the
+`instructions on GitHub <http://help.github.com/forking/>`_, which you can refer
+to for more details. This documentation includes a version specific to NEST.
+
+Creating your own forked copy of NEST
+#####################################
+
+1. Login to your GitHub account.
+2. Go to the `NEST source code repository <https://github.com/nest/nest-simulator>`_ on GitHub.
+3. Click on the ``Fork`` button.
+
+After a short pause, you should find yourself at the home page for your own
+forked copy of NEST.
+
+Downloading your fork
+#####################
+
+After forking the repository, you need to download it to your local computer to
+work with the code.
+
+The following commands should do it. The next section explains the commands.
+
+.. code::
+
+   git clone git@github.com:your-user-name/nest-simulator.git
+   cd nest-simulator
+   git remote add upstream git://github.com/nest/nest-simulator.git
+
+Commands explained
+~~~~~~~~~~~~~~~~~~
+
+**Clone your fork**
+
+.. code::
+
+  git clone git@github.com:your-user-name/nest-simulator.git
+
+This downloads your fork to your local system.  Investigate. Change directory
+to your new repository: ``cd nest-simulator``.
+Then ``git branch -a`` to show you all branches. You’ll get something like:
+
+.. code::
+
+   * master
+   remotes/origin/master
+
+This tells you that you are currently on the ``master`` branch, and that you
+also have a ``remote`` connection to ``origin/master``. The ``master`` branch is the
+default branch and this is where code that has been reviewed and tested resides. 
+``origin/master`` is just a copy of the ``master`` branch on your system on the ``remote``.
+
+What remote repository is ``remote/origin``? Try ``git remote -v`` to see the web
+address for the remote. It should point to your GitHub fork.
+
+Next, you connect your local copy to the central [NEST GitHub
+repository][NEST GitHub], so that you can keep your local copy and remote fork
```

Comment:
I've never seen this `[...][...]` syntax. What does it do? Is this a link to something? if yes, to what? Thanks!

Reply:
Ah, well spotted! This is a reference-style link in Markdown. It doesn't do anything in RST ;) I have now fixed it.